### PR TITLE
first version

### PR DESCRIPTION
**Pull request by slawekjaranowski**

### CODE REVIEW

**Comment by rrymaszewski** [comment]
test

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>5.3.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-junit-jupiter</artifactId>
+            <version>2.23.4</version>
```
Comment (rrymaszewski) [inline pom.xml]: version

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+        <!-- test -->
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>5.3.2</version>
```
Comment (rrymaszewski) [inline pom.xml]: move to dependencyManagement

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+                <groupId>org.projectlombok</groupId>
+                <artifactId>lombok</artifactId>
+                <version>1.18.6</version>
+            </dependency>
+
+            <!-- coverage -->
+            <dependency>
+                <groupId>org.apache.httpcomponents</groupId>
+                <artifactId>httpclient</artifactId>
+                <version>4.5</version>
```
Comment (rrymaszewski) [inline pom.xml:94]: extract to property

File: src/main/java/com/payu/artifactory/tools/snapshot/StorageList.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+package com.payu.artifactory.tools.snapshot;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import lombok.Getter;
+
+@Getter
+@SuppressWarnings("PMD.UnusedPrivateField") // false positive
```
Comment (rrymaszewski) [inline src/main/java/com/payu/artifactory/tools/snapshot/StorageList.java:26]: ?

File: src/main/java/com/payu/artifactory/tools/docker/DockerImageList.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+package com.payu.artifactory.tools.docker;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import lombok.Getter;
+
+@Getter
+public class DockerImageList {
```
Comment (rrymaszewski) [inline src/main/java/com/payu/artifactory/tools/docker/DockerImageList.java:27]: dockerImageList contains only list of repositories

File: src/main/java/com/payu/artifactory/tools/Config.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+@Slf4j
+public class Config {
+
+    static final String DEFAULT_PROPERTIES_FILE_PATH = "artifactory-cleaner.properties";
+
+    private final Properties properties;
+
+    public Config() {
+
+        properties = new Properties();
```
Comment (rrymaszewski) [inline src/main/java/com/payu/artifactory/tools/Config.java:38]: why in constructor?

File: src/main/java/com/payu/artifactory/tools/Config.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.Properties;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class Config {
+
+    static final String DEFAULT_PROPERTIES_FILE_PATH = "artifactory-cleaner.properties";
```
Comment (rrymaszewski) [inline src/main/java/com/payu/artifactory/tools/Config.java]: whyt it's not private?

**Review by rrymaszewski — CHANGES_REQUESTED** [review]

File: src/main/java/com/payu/artifactory/tools/snapshot/StorageList.java (hunk 4 above)
Comment (slawekjaranowski) [reply src/main/java/com/payu/artifactory/tools/snapshot/StorageList.java:26]: In public super pom we use some old version of pmd, which doesn't proper work with lombok
we can update pmd - but it is other task

File: pom.xml (hunk 3 above)
Comment (slawekjaranowski) [reply pom.xml:94]: for each version, even when we use it once?

**Review by rrymaszewski — APPROVED** [review]

**Pull request merged**